learning_paths/admin.py

```diff
@@ -115,11 +122,27 @@ class LearningPathAdmin(admin.ModelAdmin):
     def save_related(self, request, form, formsets, change):
         """Save related objects and enroll users in the learning path."""
         super().save_related(request, form, formsets, change)
+        enroll_all_in_courses = form.cleaned_data.get("enroll_all_in_courses", False)
         with transaction.atomic():
             for user in form.cleaned_data["usernames"]:
-                LearningPathEnrollment.objects.get_or_create(
-                    user=user, learning_path=form.instance
+                learning_path_enrollment, created = (
+                    LearningPathEnrollment.objects.get_or_create(
+                        user=user, learning_path=form.instance
+                    )
                 )
+                if enroll_all_in_courses:
+                    courses = LearningPathStep.objects.filter(
+                        learning_path=form.instance
+                    ).values_list("course_key", flat=True)
+                    for course_key in courses:
+                        LearningPathCourseEnrollment.objects.create(
```

Comment:
What do you think about using [bulk_create](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#bulk-create) here?